packages/terra-overlay/src/Portal.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+
+/**
+ * The <Portal/> component render its children into a div and append the div to a target DOM node.
+ */
+const propTypes = {
+  /**
+   * Whether or not the children of <Portal/> should be rendered
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * DOM element that will be appended with the new child by portal
+   */
+  target: PropTypes.object,
+};
+
+const defaultProps = {
+  isOpen: false,
+  target: document.body,
+};
+
+class Portal extends React.Component {
+  componentDidMount() {
+    if (this.props.isOpen) {
+      this.renderContent(this.props);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.isOpen) {
+      this.renderContent(nextProps);
+    } else if (this.props.isOpen && !nextProps.isOpen) {
+      this.unrenderContent();
+    }
+  }
+
+  unrenderContent() {
+    ReactDOM.unmountComponentAtNode(this.container);
+    this.props.target.removeChild(this.container);
+    this.container = null;
+  }
+
+  renderContent(props) {
+    if (!this.container) {
+      this.container = document.createElement('div');
+      props.target.appendChild(this.container);
+    }
+    ReactDOM.unstable_renderSubtreeIntoContainer(this, props.children, this.container);
+  }
+
+  render() {
+    return null;
```

Comment:
`<Portal />` itself is an empty react component. In its lifecycle it mounts/unmounts its children to other node in the DOM tree instead of the intrinsic react tree. It is like a bus. [React-bootstrap](https://github.com/react-bootstrap/react-overlays/blob/master/src/Portal.js#L92), [material-ui](https://github.com/callemall/material-ui/blob/master/src/internal/RenderToLayer.js#L120) and [react-portal](https://github.com/tajo/react-portal/blob/master/lib/portal.js#L150) have the similar implementations.